has-one.mdx

```diff
@@ -10,104 +10,78 @@ import { TipInfo } from '@/components/Tip'
-Here, users have an associated profile:
+The simplest way to specify a join for a has-one relationship is implicitly. In the example below,
+a `User` is linked to their `Profile`.
 
 ```py
 from chalk.features import features
 
 @features
 class Profile:
-    ...
-    user_id: str
+    id: str
+    user_id: "User.id"
     email_age_years: float
 
 @features
 class User:
-    ...
-    uid: Profile.user_id
+    id: str
     profile: Profile
 ```
 
-This specifies that `User.uid` is of type `str` and `User.profile` is a has-one feature
-through the join `User.uid == Profile.user_id`.
+With a has-one relationship established, you can reference features on `Profile` through
+`User`. For example:
 
+```py
+user_email_age = User.profile.email_age_years
+```
 
-### Explicit Join
+## Explicit Join
 
-The following snippet, in which the join is explicitly defined,
-is equivalent to the recommended implementation.
+In the following snippet, the has-one join is explicitly defined. This is functionally
+equivalent to the recommended implementation:
 
 ```py
 from chalk.features import features, has_one, ...
 
 @features
 class Profile:
-    ...
+    id: str
     user_id: str
     email_age_years: float
 
 @features
 class User:
-    ...
+    id: str
     uid: str
     profile: Profile = has_one(lambda: Profile.user_id == User.uid)
 ```
 
 <TipInfo>
-  The `lambda` solves forward references, letting you reference `User` before it is defined.
+  The `lambda` solves forward references, letting you reference `User` before you defined it.
 </TipInfo>
 
-You can also reference the join from the foreign feature class, if you prefer.
-The following is equivalent to the two examples above.
-
-```py
-from chalk.features import features
-
-@features
-class Profile:
-    ...
-    user_id: "User.uid"
-    email_age_years: float
-
-@features
-class User:
-    ...
-    uid: str
-    profile: Profile
-```
-
-Now, you can reference features on `Profile` through `User`. For example:
-
-```py
-user_email_age = User.profile.email_age_years
-```
-
 ---
 
 ## Back-references
 
 ### One-to-one
 
-You can also add the back-reference to `User` from `Profile`.
-You don't have to explicitly set the join on `Profile`, however.
+You can also add a back-reference to `User` from `Profile`.
+However, you don't have to explicitly set the join on `Profile`.
 Instead, the join condition is assumed to be symmetric and copied over.
-Building on the above example, all you need to do
-to complete the one-to-one relationship is to add a `User` to
-the `Profile` class:
+To complete the one-to-one relationship from our example,  add a `User`
+to the `Profile` class:
 
 <PyDiffEditor html={oneToOneDiff} />
 
 <TipInfo>Here you need to use quotes around `User` to use a forward reference.</TipInfo>
 
 ### Optional relationships
 
-You can define optional relationships simply by
-using the `typing.Optional[...]` keyword:
+You can define optional relationships by using the `typing.Optional[...]` keyword:
```

Comment:
we should define what "optional relationships" are -- i.e.:

- if a resolver takes an optional relationship, what happens?
- what happens if the local join key is set but the foreign row doesn't exist